app/smartapi.py

```diff
@@ -36,12 +40,26 @@ def get_trapi_endpoints(self):
 
         endpoints = []
         for hit in response_dict["hits"]:
+            try:
+                # check the maturity level against workflow-runner maturity
+                x_maturity = None
+                for server in hit["servers"]:
+                    if server["x-maturity"] == OPENAPI_SERVER_MATURITY:
+                        x_maturity = server["x-maturity"]
+                        break
+                if x_maturity is None:
+                    continue
+            except (KeyError):
+                continue
             try:
                 source_url = hit["_meta"]["url"]
             except (KeyError, IndexError):
                 source_url = None
             try:
-                url = hit["servers"][0]["url"]
+                for server in hit["servers"]:
```

Comment:
Since we're already looping over servers above, we could move `url` assignment up into that loop. We need to still handle the case of a missing url.